scipy/stats/tests/test_stats.py

```diff
@@ -6828,29 +6828,23 @@ def test_1d_numeric_array_like_input(self):
         gstd_actual = stats.gstd(tuple(self.array_1d))
         assert_allclose(gstd_actual, self.gstd_array_1d)
 
-    def test_raises_value_error_non_array_like_input(self):
-        with pytest.raises(ValueError, match='Invalid array input'):
-            stats.gstd('This should fail as it can not be cast to an array.')
+    def test_raises_value_error_non_numeric_input(self):
+        # this is raised by NumPy, but it's quite interpretable
+        with pytest.raises(TypeError, match="ufunc 'log' not supported"):
+            stats.gstd('You cannot take the logarithm of a string.')
 
-    def test_raises_value_error_zero_entry(self):
-        with pytest.raises(ValueError, match='Non positive value'):
-            stats.gstd(np.append(self.array_1d, [0]))
-
-    def test_raises_value_error_negative_entry(self):
-        with pytest.raises(ValueError, match='Non positive value'):
-            stats.gstd(np.append(self.array_1d, [-1]))
-
-    def test_raises_value_error_inf_entry(self):
-        with pytest.raises(ValueError, match='Infinite value'):
-            stats.gstd(np.append(self.array_1d, [np.inf]))
+    @pytest.mark.parametrize('bad_value', (0, -1, np.inf))
+    def test_returns_nan_invalid_value(self, bad_value):
+        x = np.append(self.array_1d, [bad_value])
+        assert_equal(stats.gstd(x), np.nan)
 
     def test_propagates_nan_values(self):
         a = array([[1, 1, 1, 16], [np.nan, 1, 2, 3]])
         gstd_actual = stats.gstd(a, axis=1)
         assert_allclose(gstd_actual, np.array([4, np.nan]))
 
     def test_ddof_equal_to_number_of_observations(self):
-        with pytest.raises(ValueError, match='Degrees of freedom <= 0'):
+        with pytest.warns(RuntimeWarning, match='Degrees of freedom <= 0'):
             stats.gstd(self.array_1d, ddof=self.array_1d.size)
```

Comment:
What is the outcome here? Does it return ``nan`` or a garbage value?

Reply:
According to the formula, `inf` would be the correct result here.
```suggestion
        with pytest.warns(RuntimeWarning, match='Degrees of freedom <= 0'):
            assert_equal(stats.gstd(self.array_1d, ddof=self.array_1d.size), np.inf)
```
and the test passes. Whatever it produces is consistent with `np.std`, which I think is what we'd want.